backend/trojan_dining/views.py

```diff
@@ -81,7 +83,34 @@ def post(self, request):
                 return HttpResponse(400)
             
             try:  
-                Subscription(item_id = item_id, email = email, phone_no = phone_number).save()
+                try:
+                    Subscription.objects.get(item_id = item_id, email = email, phone_no = phone_number)
+                except:
+                    Subscription(item_id = item_id, email = email, phone_no = phone_number).save()
                 return HttpResponse(201)
             except:
                 return HttpResponse(500)
+
+class GetSubscriptions(APIView):
+    def get(self, request):
+        user_email = request.GET.get('email')
+
+        if user_email is None:
+            return HttpResponse(status = 400)
```

Comment:
If i'm not mistaken, `user_email = request.GET.get('email')` will already return a 400 response if email is not provided.

If we want to set a default value to `user_email` if email is not set, then we can do `user_email = request.GET.get('email', None)`, but since we want to throw an error anyways the if statement probably isn't necessary.

Might want to double check this though haha

Reply:
changed